### DISCUSSION TITLE
How to sort DataGridTemplateColumn

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#14284] [Q&A] [answered]
I have a DataGridTemplateColumn with a button, and I would like to be able to sort based on the button Content.
I tried the following:

    <DataGridTemplateColumn Header="Error" CanUserSort="true" SortMemberPath="#errorButton.Content">
      <DataTemplate>
        <Button Name="errorButton" Content="{Binding Error}"/>
      </DataTemplate>
    </DataGridTemplateColumn>

If I click the header, a sort-triangle appears, but the grid is not being sorted.

What am I missing here?

**Comment by workgroupengineering** [comment, marked as answer]
I'm going from memory.  It should be so.

```xaml
<DataGridTemplateColumn Header="Error" CanUserSort="true" SortMemberPath="Error">
  <DataTemplate>
    <Button Name="errorButton" Content="{Binding Error}"/>
  </DataTemplate>
</DataGridTemplateColumn>

```

**Reply by wgraaf** [reply]
Thanks, it works, and it makes sense to base the sorting a property of the datacontext, not nesseccarily something present in the column.

btw. CanUserSort does not have to set:

              <DataGridTemplateColumn Header="Error" SortMemberPath="Error">
                <DataTemplate>
                  <Button Content="{Binding Error}"/>
                </DataTemplate>
              </DataGridTemplateColumn>

**Reply by Miniontoby** [reply]
Very helpful. Why isn't this documentated?

Anyways, I made a PR to add this SortMemberPath to the [DataGridTemplateColumn example/documentation](https://docs.avaloniaui.net/docs/reference/controls/datagrid/data-grid-template-columns).

https://github.com/AvaloniaUI/avalonia-docs/pull/779
